numba/tests/test_array_manipulation.py

```diff
@@ -516,6 +520,99 @@ def test_as_strided(self):
         self.check_as_strided(as_strided1)
         self.check_as_strided(as_strided2)
 
+    @unittest.skipIf(numpy_version < (1, 20), "requires NumPy 1.20 or newer")
+    def test_sliding_window_view(self):
+        cfunc = jit(nopython=True)(sliding_window_view)
+
+        def check(arr, window_shape, axis):
+            # Our version is always writeable (NumPy default is False).
+            expected = np.lib.stride_tricks.sliding_window_view(
+                arr, window_shape, axis, writeable=True
+            )
+            got = cfunc(arr, window_shape, axis)
+            self.assertPreciseEqual(got, expected)
+
+        # 1d array, different ways of specifying the axis.
+        arr1 = np.arange(24)
+        check(arr1, 5, None)
+        check(arr1, 4, 0)
+        check(arr1, 8, -1)
+        check(arr1, 11, (0,))
+
+        # 2d array, 1d window.
+        arr2 = np.arange(200).reshape(10, 20)
+        check(arr2, 5, 0)
+        check(arr2, 8, (-1,))
+
+        # 2d array, 2d window.
+        check(arr2, (4, 5), None)
+        check(arr2, (5, 8), (0, 1))
+        check(arr2, (5, 8), (1, 0))
+        check(arr2, (5, 8), (1, -2))
+
+        # 4d array, 2d window.
+        arr4 = np.arange(200).reshape(4, 5, 5, 2)
+        check(arr4, (3, 2), (1, 2))
+        check(arr4, (3, 2), (-2, -3))
+
+        # Repeated axis.
+        check(arr2, (5, 3, 3), (0, 1, 0))
+
+    @unittest.skipIf(numpy_version < (1, 20), "requires NumPy 1.20 or newer")
+    def test_sliding_window_view_errors(self):
+        cfunc = jit(nopython=True)(sliding_window_view)
+
+        def _raises(msg, *args):
+            with self.assertRaises(ValueError) as raises:
+                cfunc(*args)
+            self.assertIn(msg, str(raises.exception))
+
+        def _typing_error(msg, *args):
+            with self.assertRaises(errors.TypingError) as raises:
+                cfunc(*args)
+            self.assertIn(msg, str(raises.exception))
+
+        # Exceptions leak references
+        self.disable_leak_check()
+
+        arr1 = np.arange(24)
+        arr2 = np.arange(200).reshape(10, 20)
+
+        # Window shape validity.
+        _raises("window_shape cannot be larger", arr1, 25, None)
```

Comment:
These could be seperate subtests just so that if anything fails we know what exactly breaks on first glance. (See `TestCase.subTest`).